csrc/python_frontend/fusion_record_serde.cpp

```diff
@@ -0,0 +1,901 @@
+
+  auto ternary_alpha_tv_val_val_parser =
+      [&](const serde::RecordFunctor* buffer) {
+        return deserializeOpRecord<
+            ternary_alpha_tv_val_val_fn,
+            TensorView*,
+            TensorView*,
+            Val*,
+            Val*,
+            Val*>(
+            ternary_alpha_tv_val_val,
+            serde::RecordType_Ternary_Alpha_TV_VAL_VAL,
+            buffer);
+      };
+  registerParser(
+      serde::RecordType_Ternary_Alpha_TV_VAL_VAL,
+      ternary_alpha_tv_val_val_parser);
+
+  auto ternary_alpha_val_tv_val_parser =
+      [&](const serde::RecordFunctor* buffer) {
+        return deserializeOpRecord<
+            ternary_alpha_val_tv_val_fn,
+            TensorView*,
+            Val*,
+            TensorView*,
+            Val*,
+            Val*>(
+            ternary_alpha_val_tv_val,
+            serde::RecordType_Ternary_Alpha_VAL_TV_VAL,
+            buffer);
+      };
+  registerParser(
+      serde::RecordType_Ternary_Alpha_VAL_TV_VAL,
+      ternary_alpha_val_tv_val_parser);
+
+  auto ternary_alpha_val_parser = [&](const serde::RecordFunctor* buffer) {
+    return deserializeOpRecord<
+        ternary_alpha_val_fn,
+        Val*,
+        Val*,
+        Val*,
+        Val*,
+        Val*>(ternary_alpha_val, serde::RecordType_Ternary_Alpha_VAL, buffer);
+  };
+  registerParser(serde::RecordType_Ternary_Alpha_VAL, ternary_alpha_val_parser);
+
+  registerParser(serde::RecordType_BatchNormOp, deserializeBatchNormRecord);
+
+  registerParser(serde::RecordType_BroadcastOp, deserializeBroadcastRecord);
+
+  registerParser(
+      serde::RecordType_BroadcastInDim, deserializeBroadcastInDimRecord);
+
+  registerParser(
+      serde::RecordType_BroadcastInDimSymbolic,
+      deserializeBroadcastInDimSymbolicRecord);
+
+  registerParser(serde::RecordType_CastTv, deserializeCastTvRecord);
+
+  registerParser(serde::RecordType_CastVal, deserializeCastValRecord);
+
+  registerParser(serde::RecordType_ConstantBool, deserializeConstantBoolRecord);
+
+  registerParser(serde::RecordType_ConstantInt, deserializeConstantIntRecord);
+
+  registerParser(
+      serde::RecordType_ConstantDouble, deserializeConstantDoubleRecord);
+
+  registerParser(
+      serde::RecordType_ConstantComplexDouble,
+      deserializeConstantComplexDoubleRecord);
+
+  registerParser(serde::RecordType_FullOp, deserializeFullRecord);
+
+  registerParser(serde::RecordType_IotaOp, deserializeIotaRecord);
+
+  registerParser(serde::RecordType_TorchGatherOp, deserializeTorchGatherRecord);
+
+  registerParser(serde::RecordType_IndexSelectOp, deserializeIndexSelectRecord);
+
+  registerParser(serde::RecordType_OutputTv, deserializeOutputTvRecord);
+
+  registerParser(serde::RecordType_OutputVal, deserializeOutputValRecord);
+
+  registerParser(serde::RecordType_PermuteOp, deserializePermuteRecord);
+
+  registerParser(serde::RecordType_RandomOp, deserializeRandomRecord);
+
+  auto reduction_max_parser = [](const serde::RecordFunctor* buffer) {
```

Comment:
nitpick: what justifies the lambda here, versus defining deserialize function above?

Maybe we should just stick to one style here :laughing: .

Reply:
Originally, I used `std::bind` but clang-tidy complained. I only use lambdas when I need to capture variables, so all the parser functions have the same function signature.